rpc/types_transaction.go

```diff
@@ -93,17 +93,34 @@ func (tx L1HandlerTxn) Hash() *felt.Felt {
 	return tx.TransactionHash
 }
 
-type DeclareTxn struct {
+type DeclareTxnV1 struct {
 	CommonTransaction
 
 	// ClassHash the hash of the declared class
 	ClassHash *felt.Felt `json:"class_hash"`
 
+	DeprecatedContractClass `json:"contract_class"`
+
+	// SenderAddress the address of the account contract sending the declaration transaction
+	SenderAddress *felt.Felt `json:"sender_address"`
+}
+
+type DeclareTxnV2 struct {
```

Comment:
If there's a difference in the ContractClass property between these two types then the following is probably not necessary, but otherwise,  we could think of unifying most of the properties of DeclareTxnV1 and  DeclareTxnV2 in the same type, as most of their properties are the same

Reply:
I think we should keep them separate for now for two reasons 1) it's much easier from a users perspective to know what fields the need to fill in if they are using DeclareTxV1 and DeclareTxV2, and 2) in RPC there are three declare transaction types and merging them will cause unnecessary confusion from a dev + users perspective imo